### PR TITLE
feat: add form component

### PR DESCRIPTION
**Pull request by ryo-manba**
## 📝 Description

Added a Form component and upgraded the entire react-aria version to resolve typecheck errors from version mismatches.
The details of the react-aria update are provided in the inline comments of the following PR.
https://github.com/nextui-org/nextui/pull/3732

## ⛳️ Current behavior (updates)

Currently, there is no `Form` component available, and the version of `react-aria` does not support the latest features related to form handling.

## 🚀 New behavior

The new `Form` component has been added, allowing users to utilize features such as validation behavior switching and error handling through `serverErrors`. Additionally, the `react-aria` version has been updated to incorporate the latest form-related improvements.

https://github.com/user-attachments/assets/40f6fff6-68a0-4b92-8273-eebd7a52ae1f

## 💣 Is this a breaking change (Yes/No):

No.

## 📝 Additional Information

A guideline on how to use the new `Form` component will be added in a separate PR. 
It will be based on the following reference. 
[React Aria Forms Guide](https://react-spectrum.adobe.com/react-aria/forms.html)

## Test

Please verify the following in each story.
1. The `Form` component allows toggling of `validationBehavior`.
2. Error messages are correctly displayed based on the `name` field when provided via `serverErrors`.

## Summary by CodeRabbit

- **New Features**
  - Introduced form validation capabilities across various components including `DatePicker`, `Autocomplete`, and `Dropdown`.
  
- **Bug Fixes**
  - Enhanced error handling and validation checks in the `Dropdown` and `Autocomplete` components.
  
- **Documentation**
  - Updated stories in Storybook for `DatePicker` and `DateRangePicker` to demonstrate new server-side validation features.
  
- **Chores**
  - Updated multiple dependencies across various packages to ensure compatibility and access to the latest features and fixes.

### CODE REVIEW

**Comment by jrgarciadev** [comment]
@ryo-manba please pull the changes from canary

**Comment by jrgarciadev** [comment]
@ryo-manba fix conflicts

**Comment by ryo-manba** [comment]
@jrgarciadev I've fixed it.

**Comment by ryo-manba** [comment]
When merging, please close the related issue in this PR.
- https://github.com/nextui-org/nextui/pull/3732

**Comment by Scholar01** [comment]
Thank you for your contribution, I have been looking forward to this update for a long time!

**Comment by Scholar01** [comment]
👀👀 🙏

**Review by wingkwong — COMMENTED** [review]
seems doc & storybook failed to build. please take a look.

File: packages/core/react/package.json [hunk 1]
```diff
@@ -84,7 +84,7 @@
     "@nextui-org/date-input": "workspace:*",
     "@nextui-org/date-picker": "workspace:*",
     "@nextui-org/framer-utils": "workspace:*",
```
Comment (jrgarciadev) [inline packages/core/react/package.json:86]: @ryo-manba add the Form package here

**Comment by jrgarciadev** [comment]
@ryo-manba I uninstalled `react-aria-components` and moved only the needed types and functions to the `form` package this helps with decreasing the package size https://github.com/nextui-org/nextui/pull/3036/commits/3cf79887508b64e6a3efd4d319b51823598ee003, all tests passed

**Pull request merged**